### ISSUE TITLE
Node API: Modify Topic Handler to take Message instead of data

### ISSUE THREAD

**Issue by oskarth** [status-im/nim-waku#90]
Currently TopicHandler in subscribe looks roughly like this (nim-libp2p/pubsub.nim):

```
type TopicHandler* = proc(topic: string, data: seq[byte])
```

We want it to look like this:

```
type TopicHandler* = proc(topic: Topic, message: Message)
```

I.e. take a Message instead of data seq[byte] field

**Comment by oskarth** [comment]
Blocked by https://github.com/status-im/nim-waku/issues/92

**Comment by oskarth** [comment]
I believe this is outdated and no longer needed.
